test/Sema/sr15807.swift

```diff
@@ -0,0 +1,9 @@
+// RUN: %empty-directory(%t)
```

Comment:
A more descriptive name for the test, or a comment in the source file that explains this is about `where` clause Self bounds across module boundaries, might be good.